mlrun/api/main.py

```diff
@@ -241,7 +315,17 @@ def _cleanup_runtimes():
 
 
 def main():
-    init_data()
+    if config.httpdb.clusterization.role == mlrun.api.schemas.ClusterizationRole.chief:
+        init_data()
+    elif (
+        config.httpdb.clusterization.worker.sync_with_chief.mode
+        == mlrun.api.schemas.WaitForChiefToReachOnlineStateFeatureFlag.enabled
+        and config.httpdb.clusterization.role
+        == mlrun.api.schemas.ClusterizationRole.worker
+    ):
+        # this is being set only being set once to trigger the periodic chief state pulling
+        config.httpdb.state = mlrun.api.schemas.APIStates.waiting_for_chief
```

Comment:
I don't see this is really used, I think we can remove this state

Reply:
By default the state of the instance when starting up is online. When worker starts we don't want it to turn online by its own, only after pulling the state from the chief. 
Choosing this state and not `waiting_for_migrations` for example, will help us understand exactly in what phase the worker is. If on the first time when requesting the chief it is unable to reach it because the chief is down from some reason it is more clear that the state is `waiting_for_chief` rather than `waiting_for_migrations`. Also I don't want to set any state that isn't actually representing the true state of the cluster(that chief is dictating) because this can be confusing. If I'll set `waiting_for_migrations` but the chief state is actually online this can be confusing.

Reply:
Ok, so first of all delete/fix the comment as this is doing anything "to trigger the periodic chief state pulling"